### PR TITLE
initial manifest for org.freedesktop.dabrain34.GstPipelineStudio

### PR DESCRIPTION
**Pull request by dabrain34**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (bilelmoussaoui) [inline flathub.json:2]: is this a requirement?

File: org.freedesktop.dabrain34.GstPipelineStudio.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "--socket=fallback-x11",
+    "--socket=wayland",
+    "--device=dri",
+    "--env=G_MESSAGES_DEBUG=none",
+    "--env=RUST_BACKTRACE=1"
+  ],
+  "build-options": {
+    "append-path": "/usr/lib/sdk/rust-stable/bin",
+    "build-args": ["--share=network"],
+    "test-args": ["--socket=x11", "--share=network"]
```
Comment (bilelmoussaoui) [inline org.freedesktop.dabrain34.GstPipelineStudio.json:18]:
```suggestion
```
not needed for flathub builds

File: org.freedesktop.dabrain34.GstPipelineStudio.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  "modules": [
+    {
+      "name": "gst_pipeline_studio",
+      "buildsystem": "meson",
+      "run-tests": true,
+      "config-opts": [""],
```
Comment (bilelmoussaoui) [inline org.freedesktop.dabrain34.GstPipelineStudio.json:25]:
```suggestion
```

**Comment by bilelmoussaoui** [comment]
bot, build org.freedesktop.dabrain34.GstPipelineStudio

File: flathub.json (hunk 1 above)
Comment (dabrain34) [reply flathub.json:2]: yes true, it should work on other arch. I remove it.

File: org.freedesktop.dabrain34.GstPipelineStudio.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    "append-path": "/usr/lib/sdk/rust-stable/bin"
+  },
+  "modules": [
+    {
+      "name": "gst_pipeline_studio",
+      "buildsystem": "meson",
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://gitlab.freedesktop.org/dabrain34/GstPipelineStudio/-/archive/0.2.0/GstPipelineStudio-0.2.0.tar.gz"
```
Comment (bilelmoussaoui) [inline org.freedesktop.dabrain34.GstPipelineStudio.json:25]: There is a missing ","

**Comment by dabrain34** [comment]
bot, build org.freedesktop.dabrain34.GstPipelineStudio

File: org.freedesktop.dabrain34.GstPipelineStudio.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
+{
+  "app-id": "org.freedesktop.dabrain34.GstPipelineStudio",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "41",
+  "sdk": "org.gnome.Sdk",
+  "sdk-extensions": ["org.freedesktop.Sdk.Extension.rust-stable"],
+  "command": "gst_pipeline_studio",
+  "finish-args": [
+    "--socket=fallback-x11",
```
Comment (hfiguiere) [inline org.freedesktop.dabrain34.GstPipelineStudio.json:9]: you still need `--share=ipc`

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
cargo need to have all the package dependencies off-line. See https://github.com/flatpak/flatpak-builder-tools that has tools to do that from `Cargo.toml`.

**Comment by dabrain34** [comment]
bot, build org.freedesktop.dabrain34.GstPipelineStudio

**Pull request closed without merging**